pandas/io/common.py

```diff
@@ -713,17 +713,36 @@ def __init__(
         archive_name: Optional[str] = None,
         **kwargs,
     ):
-        if mode in ["wb", "rb"]:
-            mode = mode.replace("b", "")
+        mode = mode.replace("b", "")
         self.archive_name = archive_name
+        self.multiple_write_buffer: Optional[Union[StringIO, BytesIO]] = None
+
         kwargs_zip: Dict[str, Any] = {"compression": zipfile.ZIP_DEFLATED}
         kwargs_zip.update(kwargs)
+
         super().__init__(file, mode, **kwargs_zip)  # type: ignore[arg-type]
 
     def write(self, data):
+        # buffer multiple write calls, write on close
+        if self.multiple_write_buffer is None:
+            self.multiple_write_buffer = (
+                BytesIO() if isinstance(data, bytes) else StringIO()
+            )
+        self.multiple_write_buffer.write(data)
+
+    def _write(self) -> None:
```

Comment:
shouldn't this actually be called `.flush()`? or is that conflicting?

Reply:
I didn't think about that, flush might be a simpler solution! I will test that.

Reply:
I renamed it to `flush`, but we still have to overwrite `close`. Some tests (weirdly not all tests) were failing without overwriting `close`.